core/src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -60,14 +61,37 @@
 
     static final String DISABLE_DYNAMIC_SCRIPTING_SETTING = "script.disable_dynamic";
 
+    // a parsing function that requires a non negative int and a timevalue as arguments split by a slash
+    // this allows you to easily define rates
+    static final Function<String, Tuple<Integer, TimeValue>> MAX_COMPILATION_RATE_FUNCTION =
+            (String value) -> {
+                if (value.contains("/") == false || value.startsWith("/") || value.endsWith("/")) {
+                    throw new IllegalArgumentException("parameter must contain a positive number and a timevalue, i.e. 10/1m");
```

Comment:
This should contain the value that failed validation?

Reply:
indeed. done.